In [10]:
import pandas as pd


In [21]:
technew = pd.read_excel(r'data\Technology Skills.xlsx')     
techold = pd.read_excel(r'data\Technology Skills 260.xlsx') 
techdiff = pd.concat([technew, pd.concat([techold]*2)]).drop_duplicates(['O*NET-SOC Code','Title','Example','Commodity Code', 'Commodity Title', 'Hot Technology'], keep=False)
techdiff.shape


(1900, 6)

We see there are 1900 rows different between the 2 Technology Skills tables, comparing all the fields. This agrees with O*NET's count of differences. 
Results don't tell us which are old vs new, unfortunately. 

In [22]:
unnew = pd.read_excel(r'data\UNSPSC Reference.xlsx')
unold = pd.read_excel(r'data\UNSPSC Reference 26.0.xlsx')
undiff = pd.concat([unnew, pd.concat([unold]*2)]).drop_duplicates(['Commodity Code', 'Commodity Title',
'Class Code', 'Class Title', 'Family Code', 'Family Title', 'Segment Code', 'Segment Title'], keep=False)
undiff.shape

(416, 8)

WE see there are 416 different rows between the 2 UNSPSC tables, comparing all the fields. 

In [20]:
toolsnew = pd.read_excel(r'data\Tools Used.xlsx')     
toolsold = pd.read_excel(r'data\Tools Used 26.0.xlsx') 
toolsdiff = pd.concat([toolsnew, pd.concat([toolsold]*2)]).drop_duplicates(['O*NET-SOC Code','Title','Example','Commodity Code', 'Commodity Title'], keep=False)
toolsdiff.shape

(3127, 5)

There are 3127 different rows between the 2 tables of Tools Used, comparing all the fields. This agrees with ONET's count of differences. 

In [23]:
undiff.to_csv('output_UNSPSC_diff.csv')
techdiff.to_csv('output_Technology_Skills_diff.csv')
toolsdiff.to_csv('output_Tools_Used_diff.csv')